experimental/tools/unit-generators/vhdl/generators/support/signal_manager/__init__.py

```diff
@@ -0,0 +1,33 @@
+# See docs/Specs/SignalManager.md
+
+from collections.abc import Callable
+from .utils.types import Port, ExtraSignals
+from .normal import generate_normal_signal_manager
+from .buffered import generate_buffered_signal_manager
+from .concat import generate_concat_signal_manager
+
+
+def generate_signal_manager(name, params, generate_inner: Callable[[str], str]) -> str:
+  debugging_info = f"-- Signal manager generation info: {name}, {params}\n"
+
+  in_ports: list[Port] = params["in_ports"]
+  out_ports: list[Port] = params["out_ports"]
+  type = params["type"]
+
+  if type == "normal":
+    extra_signals: ExtraSignals = params["extra_signals"]
+    signal_manager = generate_normal_signal_manager(
+        name, in_ports, out_ports, extra_signals, generate_inner)
+  elif type == "buffered":
+    extra_signals = params["extra_signals"]
+    latency = params["latency"]
+    signal_manager = generate_buffered_signal_manager(
+        name, in_ports, out_ports, extra_signals, generate_inner, latency)
+  elif type == "concat":
+    extra_signals = params["extra_signals"]
+    signal_manager = generate_concat_signal_manager(
+        name, in_ports, out_ports, extra_signals,  generate_inner)
+  else:
+    raise ValueError(f"Unsupported signal manager type: {type}")
```

Comment:
I will remove this file and make every unit just call each signal manager in a separate PR, since it may introduce a large but unimportant diff, which might affect other people's work.